pkg/services/govmomi/service.go

```diff
@@ -213,22 +214,22 @@ func (vms *VMService) DestroyVM(ctx *context.VMContext) (reconcile.Result, infra
 	//
 
 	// Create a new virtualMachineContext to reconcile the VM.
-	vmCtx := &virtualMachineContext{
-		VMContext: *ctx,
-		Obj:       object.NewVirtualMachine(ctx.Session.Client.Client, vmRef),
+	virtualMachineCtx := &virtualMachineContext{
+		VMContext: *vmCtx,
+		Obj:       object.NewVirtualMachine(vmCtx.Session.Client.Client, vmRef),
 		Ref:       vmRef,
 		State:     &vm,
 	}
 
 	// Shut down the VM
-	powerState, err := vms.getPowerState(vmCtx)
+	powerState, err := vms.getPowerState(virtualMachineCtx)
```

Comment:
```suggestion
	powerState, err := vms.getPowerState(ctx, virtualMachineCtx)
```
Need to refactor function getPowerState

Reply:
Done